tests/test_hash.c

```diff
@@ -23,10 +23,10 @@
 /* static test data */
 const int nkeys = 5;
```

Comment:
With this patch arrays contain 6 entries. I would suggest to find number of keys in compile-time.
```suggestion
#include "test.h"
const int nkeys = ARRAY_SIZE(keys); /* move this line below the `values` array */
```